tests/e2e/run-e2e-test.sh

```diff
@@ -12,52 +12,227 @@
 
 #!/bin/bash
 
+###############################################################################
+# Set environment variables and options
+###############################################################################
+export E2E_SCENARIOS_FILE=testfiles/scenarios.yaml
+export ARRAY_INFO_FILE=array-info.sh
+export GO111MODULE=on
+export ACK_GINKGO_RC=true
+export PROG="${0}"
+
 set -o errexit
 set -o nounset
 set -o pipefail
 
-source ./env-e2e-test.sh
+PATH=$PATH:$(go env GOPATH)/bin
+
+###############################################################################
+# Function definitions
+###############################################################################
+function getArrayInfo() {
+  source ./$ARRAY_INFO_FILE
+}
+
+function checkForScenariosFile() {
+  if [ -v SCENARIOS ]; then
+    export E2E_SCENARIOS_FILE=$SCENARIOS
+  fi
 
-if [ -z "${VALUES_FILE-}" ]; then
-    echo "!!!! FATAL: VALUES_FILE was not provided"
+  stat $E2E_SCENARIOS_FILE >&/dev/null || {
+    echo "error: $E2E_SCENARIOS_FILE is not a valid scenario file - exiting"
     exit 1
-fi
-export E2E_VALUES_FILE=$VALUES_FILE
+  }
+}
 
-export GO111MODULE=on
-export ACK_GINKGO_RC=true
+function checkForCertCsi() {
+  if [ -v CERT_CSI ]; then
+    stat $CERT_CSI >&/dev/null || {
```

Comment:
Curious, why do you use stats instead of the shell built in to test executable status.